src/download/tests/download-curl-safe.rs

```diff
@@ -0,0 +1,42 @@
+#![cfg(feature = "curl-backend")]
+
+use download::*;
+
+mod support;
+use crate::support::{file_contents, serve_file, tmp_dir};
+
+/// There are two separate files because this crate caches curl handles
+/// and all tests in one file use either the safe or the unsafe handle
```

Comment:
This comment is slightly confusing, given you've made it a documentation comment but it doesn't document `downloading_with_no_certificate()` specifically, and it doesn't refer to the alternative test file.